test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/IVsServicesTestCase.cs

```diff
@@ -87,14 +86,9 @@ public void IVSPathContextProvider2_WithEmptySolution_WhenTryCreateUserWideConte
         public async Task SimpleInstallFromIVsInstaller_PackageSourceMapping_WithSingleFeed()
         {
             // Arrange
-            EnsureVisualStudioHost();
-            var dte = VisualStudio.Dte;
-            string mainDirectory = CommonUtility.CreateSolutionDirectory(Directory.GetCurrentDirectory());
-            string solutionDirectory = Path.Combine(mainDirectory, "Solution");
-            var solutionService = VisualStudio.Get<SolutionService>();
+            using var simpleTestPathContext = new SimpleTestPathContext();
             var nugetTestService = GetNuGetTestService();
-
-            var privateRepositoryPath = Path.Combine(mainDirectory, "PrivateRepository");
+            var privateRepositoryPath = Path.Combine(simpleTestPathContext.SolutionRoot, "PrivateRepository");
```

Comment:
Similar to this, you don't need it if you're using the SimpleTestPathContext. It should cover that for oyu.

Reply:
I tried to leave the test mostly alone without refactoring it too much.  I was worried about messing them up so I tried to re-use the code and use the `SimpleTestPathContext`.  Should I really get in there and re-factor these tests?

Reply:
`SimpleTestPathContext` already provides a package source: https://github.com/NuGet/NuGet.Client/blob/2e6da26086302f2c0153e026816e9b1797a0a586/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestPathContext.cs#L42

So, the test could use that, instead of the `PrivateRepository` directory.

```suggestion
            var privateRepositoryPath = simpleTestPathContext.PackageSource;
```

The following `Directory.CreateDirectory` might not be needed either. I think that SimpleTestPathContext creates it (even when it's not needed, yay slow tests!).

Reply:
I think we should fix the simple test path context, but if you don't want to do it in this PR it's ok, as long as we have a future tracking one.

I don't expect the change to be super complicated and likely to just work on the first try.

Reply:
I have refactored everything to clean things up and reduce code.  I also found some tests that were pulling newtonsoft.json from the network so now they use test packages instead.